resources/views/pages/angeltypes/qr.twig

```diff
@@ -0,0 +1,47 @@
+{% extends "layouts/app.twig" %}
+{% import 'macros/base.twig' as m %}
+{% import 'macros/form.twig' as f %}
+
+{% block title %}{{ __('angeltypes.qr', [angel_type.name]) }}{% endblock %}
+
+{% block content %}
+    <div class="container">
+
+        <h1>
+            {{ m.back(url('/angeltypes', {'action': 'view', 'angeltype_id': angel_type.id})) }}
+            {{ block('title') }}
+        </h1>
+
+        {% if qr_data %}
+            <div class="offset-md-4 col-md-4">
+                <div class="mb-3">
+                    {{ qr(qr_data, 400) }}
+                </div>
+
+                {{ f.input('url', __('general.url'), {'value': qr_data, 'readonly': true}) }}
+            </div>
+        {% endif %}
+
+        <div class="offset-md-4 col-md-4">
+            <form method="post" class="row row-cols-lg-auto g-3 align-items-center">
+                {{ csrf() }}
+
+                <div class="col-12">
```

Comment:
imho it shouldn't be possible to generate more than one code at a time. The input fields should not be visible when the QR code was generated.